### PR TITLE
[debugger] Enable reading embedded ppdb

### PR DESCRIPTION
**Pull request by thaystg**
To fix the issue of enable reading embedded pdb everywhere the problem was that when the roslyn tests are compiling sources it creates embedded pdb with source file "", and before enable reading the embedded pdb we don't read this information, and now we read and our StackTrace corlib doesn't expect a source file "", it's okay if it's NULL but not "".
I could fix in corlib or in mono, I prefer to fix on mono because I don't know if there is any other place in corlib that doesn't expect source file "".

### CODE REVIEW

File: mono/metadata/debug-mono-ppdb.c [hunk 1]
```diff
@@ -436,7 +435,8 @@ mono_ppdb_lookup_location (MonoDebugMethodInfo *minfo, uint32_t offset)
 	}
 
 	location = g_new0 (MonoDebugSourceLocation, 1);
-	location->source_file = docname;
+	if (strlen(docname) != 0)
```
Comment (jaykrell) [inline mono/metadata/debug-mono-ppdb.c]: `strlen` is very wasteful.
Just check first character.
`if (docname [0])`

File: mcs/build/config-default.make [hunk 2]
```diff
@@ -18,7 +18,7 @@ TEST_RUNTIME_FLAGS =
 endif
 
 TEST_HARNESS = $(topdir)/class/lib/$(PROFILE_DIRECTORY)/$(PARENT_PROFILE)nunit-lite-console.exe
-PLATFORM_DEBUG_FLAGS = /debug:portable
+PLATFORM_DEBUG_FLAGS = /debug:embedded
```
Comment (jaykrell) [inline mcs/build/config-default.make]: Be sure to test both/all, not delete one test to add another.

File: mcs/class/Mono.Debugger.Soft/Makefile [hunk 3]
```diff
@@ -28,7 +28,7 @@ $(test_output_dir):
 build-dtest: $(test_output_dir)/dtest-app.exe $(test_output_dir)/dtest-excfilter.exe
 
 $(test_output_dir)/dtest-excfilter.exe: Test/dtest-excfilter.il | $(test_output_dir)
-	$(ILASM) -out:$@ /exe /debug Test/dtest-excfilter.il
+	$(ILASM) -out:$@ /exe /debug:embedded Test/dtest-excfilter.il
```
Comment (jaykrell) [inline mcs/class/Mono.Debugger.Soft/Makefile]: Be sure to test both/all, not delete one test to add another.

File: mono/metadata/debug-mono-ppdb.c [hunk 4]
```diff
@@ -172,8 +172,7 @@ mono_ppdb_load_file (MonoImage *image, const guint8 *raw_contents, int size)
 		return NULL;
 	}
 
-// Temporarily disabled to unblock Roslyn
-#if HOST_WIN32 || HOST_WASM
+#if HOST_WIN32 || HAVE_SYS_ZLIB
```
Comment (jaykrell) [inline mono/metadata/debug-mono-ppdb.c]: Are embedded pdbs always compressed?

File: mcs/build/config-default.make (hunk 2 above)
Comment (jaykrell) [reply mcs/build/config-default.make]: Also, can these settings be combined -- portable and embedded?

Comment (jaykrell) [reply mcs/build/config-default.make]: Ah, embedded implies portable.

File: mono/metadata/debug-mono-ppdb.c (hunk 4 above)
Comment (thaystg) [reply mono/metadata/debug-mono-ppdb.c]: I think yes

File: mcs/class/Mono.Debugger.Soft/Makefile (hunk 3 above)
Comment (thaystg) [reply mcs/class/Mono.Debugger.Soft/Makefile]: I removed this change, I will do this in another PR.

File: mcs/build/config-default.make (hunk 2 above)
Comment (thaystg) [reply mcs/build/config-default.make]: I removed this change, I will do this in another PR.

File: mono/metadata/debug-mono-ppdb.c (hunk 1 above)
Comment (thaystg) [reply mono/metadata/debug-mono-ppdb.c]: Thanks

**Comment by thaystg** [comment]
@monojenkins build failed

File: mono/metadata/debug-mono-ppdb.c [hunk 5]
```diff
@@ -436,7 +435,8 @@ mono_ppdb_lookup_location (MonoDebugMethodInfo *minfo, uint32_t offset)
 	}
 
 	location = g_new0 (MonoDebugSourceLocation, 1);
-	location->source_file = docname;
+	if (docname != NULL && docname[0] != 0)
```
Comment (jaykrell) [inline mono/metadata/debug-mono-ppdb.c]: cosmetic: `if (docname && docname [0])`

Comment (jaykrell) [reply mono/metadata/debug-mono-ppdb.c]: or `if (!docname || docname [0])`
but yours is clearer.

Comment (thaystg) [reply mono/metadata/debug-mono-ppdb.c]: done!

**Comment by thaystg** [comment]
@monojenkins build Linux AArch64

File: mono/metadata/debug-mono-ppdb.c (hunk 4 above)
Comment (CoffeeFlux) [reply mono/metadata/debug-mono-ppdb.c]: Don't you also need the `HOST_WASM` here? I thought wasm *doesn't* have a system zlib?

**Comment by lambdageek** [comment]
This is related to #17209

**Comment by thaystg** [comment]
@monojenkins build failed

**Comment by thaystg** [comment]
@monojenkins build Linux ARMv5

**Comment by Therzok** [comment]
Is this getting backported to any current release?

**Comment by thaystg** [comment]
@monojenkins backport 2020-02

**Pull request merged**